mcs/class/WindowsBase/ZipSharp/UnzipReadStream.cs

```diff
@@ -30,6 +30,7 @@ namespace zipsharp
 {
 	class UnzipReadStream : Stream
 	{
+		private static bool use32Bit = IntPtr.Size == 4 || Environment.OSVersion.Platform != PlatformID.Unix;
```

Comment:
You could also just do all this 32/64 bit logic inside NativeUnzip then you don't need to modify all the callers.